pkg/apiclient/argo-kube-cron-workflow-service-client.go

```diff
@@ -4,35 +4,57 @@ import (
 	"context"
 
 	"google.golang.org/grpc"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	cronworkflowpkg "github.com/argoproj/argo/pkg/apiclient/cronworkflow"
 	"github.com/argoproj/argo/pkg/apis/workflow/v1alpha1"
+	"github.com/argoproj/argo/server/auth"
+	"github.com/argoproj/argo/workflow/templateresolution"
+	"github.com/argoproj/argo/workflow/validate"
 )
 
 type argoKubeCronWorkflowServiceClient struct {
-	delegate cronworkflowpkg.CronWorkflowServiceServer
```

Comment:
Please can you put this work on hold so we can sit down and discus this. Originally I implemented this similar to your change, but used the delegate pattern to simplify the code reducing bugs and maintenance overhead. For example, I'm pretty sure the code around watches and logs has some bugs in it and we would not want 2 copies of it.
I would not want to take a step back in the opposite direction by adding back a lot of complexity I'd removed.